src/Hero.java

```diff
@@ -0,0 +1,102 @@
+public class Hero extends Entity {
+
+    private int currentLevel = 1;
+    private double currentXp = 0;
+    private double xpPerLevel;
+    private int dmgIncreasePerLevel;
+    private int hpIncreasePerLevel;
+    private double cooldownMultiplierPerLevel;
+    private double originalHealth = getHealth();
+
+    public Hero(
+        double health,
+        double attack,
+        double defense,
+        double cooldown,
+        String name,
+        String lore,
+        int xpPerLevel,
+        int dmgIncreasePerLevel,
+        int hpIncreasePerLevel,
+        double cooldownMultiplierPerLevel
+        ) {
+
+        super(health, attack, defense, cooldown, name, lore);
+        
+        this.xpPerLevel=xpPerLevel;
+        this.dmgIncreasePerLevel=dmgIncreasePerLevel;
+        this.hpIncreasePerLevel=hpIncreasePerLevel;
+        this.cooldownMultiplierPerLevel=cooldownMultiplierPerLevel;
+        
+    }
+
+    public double getcooldownMultiplierPerLevel() {
+        return cooldownMultiplierPerLevel;
+    }
+
+    public void setcooldownMultiplierPerLevel(double cooldownMultiplierPerLevel) {
+        this.cooldownMultiplierPerLevel = cooldownMultiplierPerLevel;
+    }
```

Comment:
setterek maradjanak protected-ek itt is, ha feltetlenul kellenek. De szivem szerint ezek final-ok lennenek, es akkor nem. Csak ha a serializer miatt muszaj, akkor maradhat